lib/chef/knife/core/windows_bootstrap_context.rb

```diff
@@ -72,12 +58,21 @@ def trusted_certs_script
         end
 
         def config_content
+          # The windows: true / windows: false in the block that follows is more than a bit weird.  The way to read this is that we need
+          # the e.g. var_chef_dir to be rendered for the windows value ("C:\chef"), but then we are rendering into a file to be read by
+          # ruby, so we don't actually care about forward-vs-backslashes and by rendering into unix we avoid having to deal with the
+          # double-backwhacking of everything.  So we expect to see:
+          #
+          # file_cache_path "C:/chef"
+          #
+          # Which is mildly odd, but should be entirely correct as far as ruby cares.
+          #
```

Comment:
So I wrote this comment out with the explanation and I'm now I'm second guessing myself.

The issue is that ultimately the `Chef::Config[:file_cache_path]` gets used everywhere.  We could expect users to wind up trying to put this into shell_out/powershell_out commands and expect it to be rendered correctly into cmd/powershell.  If we have unix-style forward-slashes then we force users to do their own manual cleanpath'ing of this value on windows.

I'm starting to think we should eat the pain of double-backslash escaping the backslashes here...